http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java

```diff
@@ -397,6 +422,18 @@ private HttpRequest buildJettyRequest(Request finalRequest)
         jettyRequest.timeout(requestTimeoutMillis, MILLISECONDS);
         jettyRequest.idleTimeout(idleTimeoutMillis, MILLISECONDS);
 
+        // client authentications
+        if (jettyRequest.getURI().getScheme().equalsIgnoreCase("https")) {
+            jettyRequest.header(io.airlift.security.utils.SecurityUtil.REALM_IN_CHALLENGE, "true");
+            AuthenticationStore authStore = httpClient.getAuthenticationStore();
+            if (securityConfig != null && securityConfig.enabled()) {
+                String authScheme = securityConfig.getAuthScheme().toString();
+                if (authStore.findAuthentication(authScheme, jettyRequest.getURI(), null) == null) {
```

Comment:
Two things here:
1) Only the JettyHttpClient has the knowledge of which authentication scheme is supported. Having the AuthenticationStore creating the Authentication breaks the pattern and will cause the code to be less readable.
2) The current implementation of AuthenticationStore is just a workaround to improve the perf issue that Jetty has. If Jetty can fix it, we don't need it anymore.